server/hydrogen-render/1-render-hydrogen-to-string.js

```diff
@@ -85,60 +117,58 @@ async function renderHydrogenToString(renderOptions) {
         // Exited successfully
         if (exitCode === 0) {
           resolve(data);
         } else {
-          let extraErrorsMessage = '';
-          if (childErrors.length > 1) {
-            extraErrorsMessage = ` (somehow we saw ${
-              childErrors.length
-            } errors but we really always expect 1 error)\n${childErrors
-              .map((childError, index) => `${index}. ${childError.stack}`)
-              .join('\n')}`;
-          }
-
-          let childErrorToDisplay = new Error('No child errors');
-          if (childErrors.length === 1) {
-            childErrorToDisplay = childErrors[0];
-          } else if (childErrors.length > 1) {
-            childErrorToDisplay = new Error('Multiple child errors listed above ^');
-          }
-
-          const error = new RethrownError(
-            `Child process failed with exit code ${exitCode}${extraErrorsMessage}`,
-            childErrorToDisplay
+          const childErrorSummary = assembleErrorAfterChildExitsWithErrors(
+            childExitCode,
+            childErrors
           );
-          reject(error);
+          reject(childErrorSummary);
         }
       });
 
       // When a problem occurs when spawning the process or gets aborted
       child.on('error', (err) => {
         if (err.name === 'AbortError') {
-          throw new RethrownError(
-            `Timed out while rendering Hydrogen to string so we aborted the child process after ${RENDER_TIMEOUT}ms`,
-            err
+          const childErrorSummary = assembleErrorAfterChildExitsWithErrors(
+            childExitCode,
+            childErrors
+          );
+          reject(
+            new RethrownError(
+              `Timed out while rendering Hydrogen to string so we aborted the child process after ${RENDER_TIMEOUT}ms. Any child errors? (${childErrors.length})`,
+              childErrorSummary
+            )
           );
+        } else {
+          reject(err);
         }
-
-        reject(err);
       });
     });
 
-    assert(
-      data,
-      `No HTML sent from child process to render Hydrogen. Any child errors? (${childErrors.length})`
-    );
+    if (!returnedData) {
+      const childErrorSummary = assembleErrorAfterChildExitsWithErrors(childExitCode, childErrors);
+      throw new RethrownError(
+        `No HTML sent from child process to render Hydrogen. Any child errors? (${childErrors.length})`,
+        childErrorSummary
+      );
+    }
 
-    return data;
+    return returnedData;
   } catch (err) {
     throw new RethrownError(
       `Failed to render Hydrogen to string. In order to reproduce, feed in these arguments into \`renderHydrogenToString(...)\`:\n    renderToString arguments: ${JSON.stringify(
         renderOptions
       )}`,
       err
     );
+  } finally {
+    // We don't have to add a undefined/null check here because `clearTimeout`
+    // works with any value you give it and doesn't throw an error.
+    clearTimeout(abortTimeoutId);
```

Comment:
This wasn't strictly necessary but it's good practice to stop the timeout callback from aborting nothing later.